### PR TITLE
[IMP] *: Editing currencies display symbol for better differentiation

### PR DESCRIPTION
**Pull request by Megaaaaaa**
*: account, base, base_import, hr_expense, mrp_subcontracting_account,
product, purchase, purchase_stock, sale, spreadsheet, stock_account,
test_xlsx_export, web, website_event_sale

Description of the issue/feature this PR addresses:

Prior to this, when multiple currencies with the same symbol where to appear
in a same document or be sent from a country to another with the same currency
symbol, the only information the recipient had was
the currency symbol leading to lack of clarity.

Desired behavior after PR is merged:

This implementation makes it now automatic that when multiple currencies with
the same symbol are active, it is their ISO code instead of their symbol
that is used to display them. This way, there is no room for misinterpretation.

Whenever a currency is activated, this code checks whether or not another
currency with the same symbol is already active. If it is the case, the ISO
code replaces the symbol of the newly activated currency. Also, when a currency
is deactivated, this code checks whether or not a single currency with the same
symbol is still active. If it is the case, it puts the symbol of that currency
back to its original state. If not, it means that there are still multiple
currencies with that symbol active in which case their symbol should
not change. In both cases, the symbol of the currencies which is being 
deactivated is reversed to its original value, the currency's symbol.

The features interacts with three fields. Each of them is on the res.currency
model. The field "name" which is used to hold the currency's ISO code. No
change has been made to that field. The field "curr_symbol" which is new and 
takes the field "symbol" previous role which was to hold the currency's symbol.
And the field "symbol" which is now a computed field which value will depend
on the number of active currencies with the same symbol as the one
the computation is happening on.

Adding this improvement, the differentiation between currencies with a similar
symbol is now way clearer.

task-3360250

### CODE REVIEW

File: odoo/addons/base/models/res_currency.py [hunk 1]
```diff
@@ -323,6 +324,22 @@ def _get_view(self, view_id=None, view_type='form', **options):
                     node[0].set('string', field[1])
         return arch, view
 
+    @api.onchange('active')
```
Comment (ryv-odoo) [inline odoo/addons/base/models/res_currency.py:327]: `onchange` is not suitable for this:
- It is only called during changes of the form view, but it should be also called during installation of module (or others modifications).
- `onchange` shouldn't not modify the database for many reasons (by example 'discard' button won't discard change in database which lead your database into an inconsistency state)

Override write/create/unlink instead, it is much more appropriate.

File: odoo/addons/base/models/res_currency.py [hunk 2]
```diff
@@ -323,6 +324,22 @@ def _get_view(self, view_id=None, view_type='form', **options):
...
+                for currency in active_currencies_same_symbol:
+                    currency.symbol = currency.name
+            elif len(active_currencies_same_symbol) > 1:
+                self.symbol = self.name
+        else:
+            if len(active_currencies_same_symbol) == 2:
+                for currency in active_currencies_same_symbol:
+                    currency.symbol = currency.non_unique_symbol
+            elif len(active_currencies_same_symbol) > 2:
+                self.symbol = self.non_unique_symbol
```
Comment (ryv-odoo) [inline odoo/addons/base/models/res_currency.py:342]: it is an important feature, can you add test for each scenario?

File: odoo/addons/base/models/res_currency.py (hunk 1 above)
Comment (Julien00859) [reply odoo/addons/base/models/res_currency.py:327]: I'm +1 for a test, right now I have an hard time understanding what's going on and I'm not even sure this code is right

**Review by Julien00859 — COMMENTED** [review]
So now the `symbol` column kind of acts like a computational field, either taking the `name` (ISO code), either the `non_unique_symbol` column as value. In the XML is was necessary to duplicate every `symbol` entry into `non_unique_symbol` just for the sake of having the original currency symbol saved somewhere (so it can be restored in case one of two active same-symbol-currency is archived).

I don't like it.

In my opinion we should make a new field e.g. `display_symbol` that does the computation name or symbol and we should edit the various templates to replace `symbol` by `display_symbol`

**Comment by fpodoo** [comment]
We should cancel this PR; this is not what we want. Currency information is highly contextual;
- if I get an invoice from an american company, it's obvious that $ means USD
- it's not because you open a small subsidiary in New Zealand that an american company want to use USD instead of $ everywhere

It's better to let the user keep the control and decide what they want, example:
- Use $ for both, and let the context imply the meaning
- Use NZD for New Zealand, but keep $ for USA
- Use NZD & USD

No need to add complexity (two currency field: curr_symbol & symbol), the current behaviour is what most people expect.

**Pull request closed without merging**